tests/bettertransformer/testing_utils.py

```diff
@@ -89,32 +89,34 @@ def _test_fp16_inference(self, model_id: str, use_to_operator=False, **preproces
 
         torch.manual_seed(0)
         if not use_to_operator:
-            hf_random_model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=torch.float16).to(0)
+            hf_random_model = automodel_class.from_pretrained(model_id, torch_dtype=torch.float16).to(0)
+            converted_model = BetterTransformer.transform(hf_random_model, keep_original_model=True)
         else:
-            hf_random_model = AutoModelForCausalLM.from_pretrained(model_id).to(0, torch.float16)
-
-        torch.manual_seed(0)
-        converted_model = BetterTransformer.transform(hf_random_model, keep_original_model=True)
+            hf_random_model = automodel_class.from_pretrained(model_id).to(0)
+            converted_model = BetterTransformer.transform(hf_random_model, keep_original_model=True)
+            hf_random_model = hf_random_model.to(torch.float16)
+            converted_model = converted_model.to(torch.float16)
 
         self.assertFalse(
             hasattr(hf_random_model, "use_bettertransformer"),
             f"The model {hf_random_model.__class__.__name__} has been converted to a `fast` model by mistake.",
         )
 
+        length = 50
         with torch.no_grad():
             r"""
             Make sure the models are in eval mode! Make also sure that the original model
             has not been converted to a fast model. The check is done above.
             """
             torch.manual_seed(0)
-            output_hf = hf_random_model.generate(**inputs)
+            output_hf = hf_random_model.generate(**inputs, min_length=length, max_length=length)
```

Comment:
I don't really see the advantage of using `auto_model_class` since here we always call `generate`

Reply:
Makes sense! Thanks for clarifying